### PR TITLE
60 fix url checker

### PR DESCRIPTION
**Pull request by jwalton3141**
Addresses #60

### CODE REVIEW

**Comment by jwalton3141** [comment]
Notes from @csgillespie's slack messages:

> For info, all `url_exists` does is check that the response status is 2**
> 
> I'm not keen on using `Rcurl::url_exists()`. It no longer displays the status code which is useful. Additionally, it's another dependency

File: R/check_urls.R [hunk 1]
```diff
@@ -15,37 +16,30 @@ check_urls = function() {
...
+          args = c("-i", shQuote("/^\\FunctionTok{git}.*/d ; s/{-}/-/g"),
+                   "extractor-tmp.tex"))
 
-    if (class(ping) == "try-error") {
-      msg_warning(glue("{url}: {ping}"), padding = TRUE)
+  grepped_url = suppressWarnings(system2("grep",
+                                         c('-Eo "(http|https)://[a-zA-Z0-9./?=_-]*"', "extractor-tmp.tex"), #nolint
+                                         stdout = TRUE))
+  urls = unique(c(urls,  grepped_url))
+  url_statuses = RCurl::url.exists(urls)
```
Comment (jwalton3141) [inline R/check_urls.R:29]: From @csgillespie's slack messages:

> What about:
> ```
> url_exists = function(url) {
>   ping = try(httr::GET(url), silent = TRUE)
>   if (class(ping) == "try-error") {
>     status = ping[1]
>     url_exists = FALSE
>   } else {
>     status = ping$status
>     url_exists = ping$status %in% 200:205
>   }
>   list(url_exists, url, status)
> }
> url_exists("jumpingrivers.com")
> url_exists("dsfsdf.com")
> ```

Comment (csgillespie) [reply R/check_urls.R:29]: @jwalton3141 @trianglegirl Remember that any url correctly labels in the .Rmd file is listed (correctly) in extractor.csv

Comment (jwalton3141) [reply R/check_urls.R:29]: I didn't know about `extractor.csv`, but this looks like the way to go.

Comment (jwalton3141) [reply R/check_urls.R:29]: @csgillespie re. your `url_exists()`, I get different output to what you indicated you got in slack:

```
r$> url_exists("dsfsdf.com")
[[1]]
[1] TRUE

[[2]]
[1] "dsfsdf.com"

[[3]]
[1] 200
```

In fact, any nonsense URL I check returns a 200 status (???)

**Comment by csgillespie** [comment]
I have a different DNS server that blocks dodgy, spammy domains - hence the
odd response

File: R/check_urls.R (hunk 1 above)
Comment (jwalton3141) [reply R/check_urls.R:29]: I think  `url_exists()` function could maybe be replaced by `httr::http_status(GET())`? Does pretty much the same but gives a human readable status code

Comment (csgillespie) [reply R/check_urls.R:29]: I have a different DNS server that blocks dodgy, spammy domains - hence the odd response

Comment (csgillespie) [reply R/check_urls.R:29]: For `url_exisits` try
```
url_exists("jumpingrivers1.com")
url_exists("jumpingrivers.com/fake")
```

File: R/check_urls.R [hunk 2]
```diff
@@ -1,45 +1,48 @@
...
+  if (all(as.integer(df_status[,"exists"]))) {
     msg_success("URLs look good")
   } else {
     msg_error("Fix broken URLS")
   }
   return(invisible(NULL))
 }
+
+url_exists = function(url) {
+  ping = try(httr::HEAD(url), silent = TRUE)
```
Comment (jwalton3141) [inline R/check_urls.R:39]: Only fetching the HEAD should be quicker than a full GET

**Comment by jwalton3141** [comment]
Closing and moving to #62 as it was getting a bit busy here...

**Pull request closed without merging**